text/000-rustdoc-jump-to-definition.md

```diff
@@ -0,0 +1,187 @@
+
+The same applies when reading a source code when type annotations are not available:
+
+```
+let var = some_other_var.do_something();
+let var2 = a_function();
+```
+
+It would require to look for each method in the various files, etc.
+
+So overall, this would greatly improve the source code browsing experience.
+
+The other goal is to allow to go back from the source code pages directly to the item documentation to be able to read the rendered content and not just the "raw" doc comments.
+
+So in short, this feature is split in two parts:
+
+ * Jump to the definition of an ident.
+ * Jump to an item documentation from the source view.
+
+Adding this information directly into the documentation allows it to improve the documentation value: if you want to go a bit further what is written in the documentation (to see some implementation details for example), you can currently go to the source code. However, you're quickly limited to the current file if it's using private items not from this file.
+
+Once you found what you're looking for, having a link to an item's documentation from the source code allows to add the missing connection between the source code viewer and the doc pages.
+
+Another important note is that if documentation is split in multiple places or requires an external tool for the source code navigation, it's actually slowing down our users quite a lot.
+
+A good example where this would be very helpful is for the [rust compiler source code](https://doc.rust-lang.org/1.57.0/nightly-rustc/src/rustc_middle/hir/map/mod.rs.html#872-877): even if a fully set up IDE or github, it's very hard to go around without previous knowledge or help from someone else. It's also very common to have undocumented items or to just want to see how something is implemented. With the "jump to definition", it already improved quite a lot the browsing experience by allowing to jump super quickly by simply clicking on a item. The missing part is now to go back from the source code into the item documentation to make it complete.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+It doesn't require javascript since it's only generating links, making it work with or without javascript enabled.
+
+For the link generation, we use the already existing system in rustdoc. Meaning their generation will follow the same rules as the link in the documentation pages.
+
+All items with documentation should have a link to their documentation page (like impl blocks, methods, trait/impl associated items, etc).
+
+So on the technical side, there is no big issues (based on what is described in this RFC). A few parts of this RFC are implemented in the following pull requests:
```

Comment:
This sentence is still completely out of place, an RFC does not get to pre-claim "there are no big issues". The point of the RFC process is to figure out there are no big issues.

Reply:
Ok, I'll remove it entirely then.